apps/web/src/app/api/uploadthing/core.ts

```diff
@@ -0,0 +1,31 @@
+import { createUploadthing, type FileRouter } from 'uploadthing/next';
+import { UploadThingError } from 'uploadthing/server';
+
+const f = createUploadthing();
+
+const auth = (req: Request) => ({ id: 'fakeId' }); // TODO: replace to real auth validation
+
+export const ourFileRouter = {
+  imageUploader: f({
+    image: {
+      maxFileSize: '4MB',
+      maxFileCount: 1,
+    },
+  })
+    .middleware(async ({ req }) => {
+      const user = await auth(req);
+
+      if (!user) throw new UploadThingError('Unauthorized');
+
+      return { userId: user.id };
+    })
+    .onUploadComplete(async ({ metadata, file }) => {
+      console.log('Upload complete for userId:', metadata.userId);
```

Comment:
In this method, you can call into the service layer, updating the corresponding entity.

Reply:
I think there is no need for this, since useSWR fetches data quite frequently, and immediately loads new (updated) data after creation/editing. What do you think? Maybe I misunderstood you.